{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "source": [
    "!pip install taxonerd"
   ],
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "Requirement already satisfied: taxonerd in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (1.1.1)\n",
      "Requirement already satisfied: pandas>=0.24.2 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from taxonerd) (1.3.2)\n",
      "Requirement already satisfied: spacy[transformers]<3.1.0,>=3.0.5 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from taxonerd) (3.0.7)\n",
      "Requirement already satisfied: click in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from taxonerd) (7.1.2)\n",
      "Requirement already satisfied: scispacy==0.4.0 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from taxonerd) (0.4.0)\n",
      "Requirement already satisfied: textract>=1.6.3 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from taxonerd) (1.6.4)\n",
      "Requirement already satisfied: pysbd in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from scispacy==0.4.0->taxonerd) (0.3.4)\n",
      "Requirement already satisfied: joblib in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from scispacy==0.4.0->taxonerd) (1.0.1)\n",
      "Requirement already satisfied: requests<3.0.0,>=2.0.0 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from scispacy==0.4.0->taxonerd) (2.26.0)\n",
      "Requirement already satisfied: conllu in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from scispacy==0.4.0->taxonerd) (4.4.1)\n",
      "Requirement already satisfied: nmslib>=1.7.3.6 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from scispacy==0.4.0->taxonerd) (2.1.1)\n",
      "Requirement already satisfied: numpy in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from scispacy==0.4.0->taxonerd) (1.21.2)\n",
      "Requirement already satisfied: scikit-learn>=0.20.3 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from scispacy==0.4.0->taxonerd) (0.24.2)\n",
      "Requirement already satisfied: psutil in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from nmslib>=1.7.3.6->scispacy==0.4.0->taxonerd) (5.8.0)\n",
      "Requirement already satisfied: pybind11<2.6.2 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from nmslib>=1.7.3.6->scispacy==0.4.0->taxonerd) (2.6.1)\n",
      "Requirement already satisfied: pytz>=2017.3 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from pandas>=0.24.2->taxonerd) (2021.1)\n",
      "Requirement already satisfied: python-dateutil>=2.7.3 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from pandas>=0.24.2->taxonerd) (2.8.2)\n",
      "Requirement already satisfied: six>=1.5 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from python-dateutil>=2.7.3->pandas>=0.24.2->taxonerd) (1.12.0)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from requests<3.0.0,>=2.0.0->scispacy==0.4.0->taxonerd) (2021.5.30)\n",
      "Requirement already satisfied: charset-normalizer~=2.0.0 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from requests<3.0.0,>=2.0.0->scispacy==0.4.0->taxonerd) (2.0.0)\n",
      "Requirement already satisfied: idna<4,>=2.5 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from requests<3.0.0,>=2.0.0->scispacy==0.4.0->taxonerd) (3.1)\n",
      "Requirement already satisfied: urllib3<1.27,>=1.21.1 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from requests<3.0.0,>=2.0.0->scispacy==0.4.0->taxonerd) (1.26.6)\n",
      "Requirement already satisfied: threadpoolctl>=2.0.0 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from scikit-learn>=0.20.3->scispacy==0.4.0->taxonerd) (2.2.0)\n",
      "Requirement already satisfied: scipy>=0.19.1 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from scikit-learn>=0.20.3->scispacy==0.4.0->taxonerd) (1.7.1)\n",
      "Requirement already satisfied: thinc<8.1.0,>=8.0.3 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from spacy[transformers]<3.1.0,>=3.0.5->taxonerd) (8.0.10)\n",
      "Requirement already satisfied: jinja2 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from spacy[transformers]<3.1.0,>=3.0.5->taxonerd) (3.0.1)\n",
      "Requirement already satisfied: typer<0.4.0,>=0.3.0 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from spacy[transformers]<3.1.0,>=3.0.5->taxonerd) (0.3.2)\n",
      "Requirement already satisfied: preshed<3.1.0,>=3.0.2 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from spacy[transformers]<3.1.0,>=3.0.5->taxonerd) (3.0.5)\n",
      "Requirement already satisfied: cymem<2.1.0,>=2.0.2 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from spacy[transformers]<3.1.0,>=3.0.5->taxonerd) (2.0.5)\n",
      "Requirement already satisfied: catalogue<2.1.0,>=2.0.4 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from spacy[transformers]<3.1.0,>=3.0.5->taxonerd) (2.0.5)\n",
      "Requirement already satisfied: tqdm<5.0.0,>=4.38.0 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from spacy[transformers]<3.1.0,>=3.0.5->taxonerd) (4.62.2)\n",
      "Requirement already satisfied: setuptools in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from spacy[transformers]<3.1.0,>=3.0.5->taxonerd) (58.0.4)\n",
      "Requirement already satisfied: blis<0.8.0,>=0.4.0 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from spacy[transformers]<3.1.0,>=3.0.5->taxonerd) (0.7.4)\n",
      "Requirement already satisfied: spacy-legacy<3.1.0,>=3.0.5 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from spacy[transformers]<3.1.0,>=3.0.5->taxonerd) (3.0.8)\n",
      "Requirement already satisfied: murmurhash<1.1.0,>=0.28.0 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from spacy[transformers]<3.1.0,>=3.0.5->taxonerd) (1.0.5)\n",
      "Requirement already satisfied: srsly<3.0.0,>=2.4.1 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from spacy[transformers]<3.1.0,>=3.0.5->taxonerd) (2.4.1)\n",
      "Requirement already satisfied: wasabi<1.1.0,>=0.8.1 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from spacy[transformers]<3.1.0,>=3.0.5->taxonerd) (0.8.2)\n",
      "Requirement already satisfied: packaging>=20.0 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from spacy[transformers]<3.1.0,>=3.0.5->taxonerd) (21.0)\n",
      "Requirement already satisfied: pathy>=0.3.5 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from spacy[transformers]<3.1.0,>=3.0.5->taxonerd) (0.6.0)\n",
      "Requirement already satisfied: pydantic!=1.8,!=1.8.1,<1.9.0,>=1.7.4 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from spacy[transformers]<3.1.0,>=3.0.5->taxonerd) (1.8.2)\n",
      "Requirement already satisfied: pyparsing>=2.0.2 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from packaging>=20.0->spacy[transformers]<3.1.0,>=3.0.5->taxonerd) (2.4.7)\n",
      "Requirement already satisfied: smart-open<6.0.0,>=5.0.0 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from pathy>=0.3.5->spacy[transformers]<3.1.0,>=3.0.5->taxonerd) (5.2.1)\n",
      "Requirement already satisfied: typing-extensions>=3.7.4.3 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from pydantic!=1.8,!=1.8.1,<1.9.0,>=1.7.4->spacy[transformers]<3.1.0,>=3.0.5->taxonerd) (3.10.0.0)\n",
      "Requirement already satisfied: spacy-transformers<1.1.0,>=1.0.1 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from spacy[transformers]<3.1.0,>=3.0.5->taxonerd) (1.0.4)\n",
      "Requirement already satisfied: torch>=1.5.0 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from spacy-transformers<1.1.0,>=1.0.1->spacy[transformers]<3.1.0,>=3.0.5->taxonerd) (1.9.0)\n",
      "Requirement already satisfied: transformers<4.10.0,>=3.4.0 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from spacy-transformers<1.1.0,>=1.0.1->spacy[transformers]<3.1.0,>=3.0.5->taxonerd) (4.9.2)\n",
      "Requirement already satisfied: spacy-alignments<1.0.0,>=0.7.2 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from spacy-transformers<1.1.0,>=1.0.1->spacy[transformers]<3.1.0,>=3.0.5->taxonerd) (0.8.3)\n",
      "Requirement already satisfied: xlrd~=1.2.0 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from textract>=1.6.3->taxonerd) (1.2.0)\n",
      "Requirement already satisfied: extract-msg<=0.29.* in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from textract>=1.6.3->taxonerd) (0.28.7)\n",
      "Requirement already satisfied: python-pptx~=0.6.18 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from textract>=1.6.3->taxonerd) (0.6.19)\n",
      "Requirement already satisfied: pdfminer.six==20191110 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from textract>=1.6.3->taxonerd) (20191110)\n",
      "Requirement already satisfied: SpeechRecognition~=3.8.1 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from textract>=1.6.3->taxonerd) (3.8.1)\n",
      "Requirement already satisfied: docx2txt~=0.8 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from textract>=1.6.3->taxonerd) (0.8)\n",
      "Requirement already satisfied: beautifulsoup4~=4.8.0 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from textract>=1.6.3->taxonerd) (4.8.2)\n",
      "Requirement already satisfied: chardet==3.* in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from textract>=1.6.3->taxonerd) (3.0.4)\n",
      "Requirement already satisfied: EbookLib==0.* in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from textract>=1.6.3->taxonerd) (0.17.1)\n",
      "Requirement already satisfied: argcomplete~=1.10.0 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from textract>=1.6.3->taxonerd) (1.10.3)\n",
      "Requirement already satisfied: lxml in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from EbookLib==0.*->textract>=1.6.3->taxonerd) (4.6.3)\n",
      "Requirement already satisfied: pycryptodome in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from pdfminer.six==20191110->textract>=1.6.3->taxonerd) (3.10.1)\n",
      "Requirement already satisfied: sortedcontainers in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from pdfminer.six==20191110->textract>=1.6.3->taxonerd) (2.4.0)\n",
      "Requirement already satisfied: soupsieve>=1.2 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from beautifulsoup4~=4.8.0->textract>=1.6.3->taxonerd) (2.2.1)\n",
      "Requirement already satisfied: imapclient==2.1.0 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from extract-msg<=0.29.*->textract>=1.6.3->taxonerd) (2.1.0)\n",
      "Requirement already satisfied: compressed-rtf>=1.0.6 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from extract-msg<=0.29.*->textract>=1.6.3->taxonerd) (1.0.6)\n",
      "Requirement already satisfied: olefile>=0.46 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from extract-msg<=0.29.*->textract>=1.6.3->taxonerd) (0.46)\n",
      "Requirement already satisfied: tzlocal>=2.1 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from extract-msg<=0.29.*->textract>=1.6.3->taxonerd) (3.0)\n",
      "Requirement already satisfied: ebcdic>=1.1.1 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from extract-msg<=0.29.*->textract>=1.6.3->taxonerd) (1.1.1)\n",
      "Requirement already satisfied: XlsxWriter>=0.5.7 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from python-pptx~=0.6.18->textract>=1.6.3->taxonerd) (3.0.1)\n",
      "Requirement already satisfied: Pillow>=3.3.2 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from python-pptx~=0.6.18->textract>=1.6.3->taxonerd) (8.3.2)\n",
      "Requirement already satisfied: pyyaml>=5.1 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from transformers<4.10.0,>=3.4.0->spacy-transformers<1.1.0,>=1.0.1->spacy[transformers]<3.1.0,>=3.0.5->taxonerd) (5.4.1)\n",
      "Requirement already satisfied: huggingface-hub==0.0.12 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from transformers<4.10.0,>=3.4.0->spacy-transformers<1.1.0,>=1.0.1->spacy[transformers]<3.1.0,>=3.0.5->taxonerd) (0.0.12)\n",
      "Requirement already satisfied: filelock in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from transformers<4.10.0,>=3.4.0->spacy-transformers<1.1.0,>=1.0.1->spacy[transformers]<3.1.0,>=3.0.5->taxonerd) (3.0.12)\n",
      "Requirement already satisfied: sacremoses in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from transformers<4.10.0,>=3.4.0->spacy-transformers<1.1.0,>=1.0.1->spacy[transformers]<3.1.0,>=3.0.5->taxonerd) (0.0.45)\n",
      "Requirement already satisfied: tokenizers<0.11,>=0.10.1 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from transformers<4.10.0,>=3.4.0->spacy-transformers<1.1.0,>=1.0.1->spacy[transformers]<3.1.0,>=3.0.5->taxonerd) (0.10.3)\n",
      "Requirement already satisfied: regex!=2019.12.17 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from transformers<4.10.0,>=3.4.0->spacy-transformers<1.1.0,>=1.0.1->spacy[transformers]<3.1.0,>=3.0.5->taxonerd) (2021.8.28)\n",
      "Requirement already satisfied: MarkupSafe>=2.0 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from jinja2->spacy[transformers]<3.1.0,>=3.0.5->taxonerd) (2.0.1)\n"
     ]
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "markdown",
   "source": [
    "Install NER models"
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "source": [
    "!pip install https://github.com/nleguillarme/taxonerd/releases/download/v1.1.1/en_ner_eco_md-1.0.0.tar.gz\n",
    "!pip install https://github.com/nleguillarme/taxonerd/releases/download/v1.1.1/en_ner_eco_biobert-1.0.0.tar.gz\n"
   ],
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "Collecting https://github.com/nleguillarme/taxonerd/releases/download/v1.1.1/en_ner_eco_md-1.0.0.tar.gz\n",
      "  Downloading https://github.com/nleguillarme/taxonerd/releases/download/v1.1.1/en_ner_eco_md-1.0.0.tar.gz (123.3 MB)\n",
      "\u001b[K     |████████████████████████████████| 123.3 MB 154 kB/s \n",
      "\u001b[?25hRequirement already satisfied: spacy<3.1.0,>=3.0.5 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from en-ner-eco-md==1.0.0) (3.0.7)\n",
      "Requirement already satisfied: pydantic!=1.8,!=1.8.1,<1.9.0,>=1.7.4 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from spacy<3.1.0,>=3.0.5->en-ner-eco-md==1.0.0) (1.8.2)\n",
      "Requirement already satisfied: jinja2 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from spacy<3.1.0,>=3.0.5->en-ner-eco-md==1.0.0) (3.0.1)\n",
      "Requirement already satisfied: tqdm<5.0.0,>=4.38.0 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from spacy<3.1.0,>=3.0.5->en-ner-eco-md==1.0.0) (4.62.2)\n",
      "Requirement already satisfied: wasabi<1.1.0,>=0.8.1 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from spacy<3.1.0,>=3.0.5->en-ner-eco-md==1.0.0) (0.8.2)\n",
      "Requirement already satisfied: typer<0.4.0,>=0.3.0 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from spacy<3.1.0,>=3.0.5->en-ner-eco-md==1.0.0) (0.3.2)\n",
      "Requirement already satisfied: blis<0.8.0,>=0.4.0 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from spacy<3.1.0,>=3.0.5->en-ner-eco-md==1.0.0) (0.7.4)\n",
      "Requirement already satisfied: setuptools in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from spacy<3.1.0,>=3.0.5->en-ner-eco-md==1.0.0) (58.0.4)\n",
      "Requirement already satisfied: spacy-legacy<3.1.0,>=3.0.5 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from spacy<3.1.0,>=3.0.5->en-ner-eco-md==1.0.0) (3.0.8)\n",
      "Requirement already satisfied: catalogue<2.1.0,>=2.0.4 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from spacy<3.1.0,>=3.0.5->en-ner-eco-md==1.0.0) (2.0.5)\n",
      "Requirement already satisfied: numpy>=1.15.0 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from spacy<3.1.0,>=3.0.5->en-ner-eco-md==1.0.0) (1.21.2)\n",
      "Requirement already satisfied: pathy>=0.3.5 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from spacy<3.1.0,>=3.0.5->en-ner-eco-md==1.0.0) (0.6.0)\n",
      "Requirement already satisfied: thinc<8.1.0,>=8.0.3 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from spacy<3.1.0,>=3.0.5->en-ner-eco-md==1.0.0) (8.0.10)\n",
      "Requirement already satisfied: srsly<3.0.0,>=2.4.1 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from spacy<3.1.0,>=3.0.5->en-ner-eco-md==1.0.0) (2.4.1)\n",
      "Requirement already satisfied: cymem<2.1.0,>=2.0.2 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from spacy<3.1.0,>=3.0.5->en-ner-eco-md==1.0.0) (2.0.5)\n",
      "Requirement already satisfied: packaging>=20.0 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from spacy<3.1.0,>=3.0.5->en-ner-eco-md==1.0.0) (21.0)\n",
      "Requirement already satisfied: requests<3.0.0,>=2.13.0 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from spacy<3.1.0,>=3.0.5->en-ner-eco-md==1.0.0) (2.26.0)\n",
      "Requirement already satisfied: murmurhash<1.1.0,>=0.28.0 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from spacy<3.1.0,>=3.0.5->en-ner-eco-md==1.0.0) (1.0.5)\n",
      "Requirement already satisfied: preshed<3.1.0,>=3.0.2 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from spacy<3.1.0,>=3.0.5->en-ner-eco-md==1.0.0) (3.0.5)\n",
      "Requirement already satisfied: pyparsing>=2.0.2 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from packaging>=20.0->spacy<3.1.0,>=3.0.5->en-ner-eco-md==1.0.0) (2.4.7)\n",
      "Requirement already satisfied: smart-open<6.0.0,>=5.0.0 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from pathy>=0.3.5->spacy<3.1.0,>=3.0.5->en-ner-eco-md==1.0.0) (5.2.1)\n",
      "Requirement already satisfied: typing-extensions>=3.7.4.3 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from pydantic!=1.8,!=1.8.1,<1.9.0,>=1.7.4->spacy<3.1.0,>=3.0.5->en-ner-eco-md==1.0.0) (3.10.0.0)\n",
      "Requirement already satisfied: idna<4,>=2.5 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from requests<3.0.0,>=2.13.0->spacy<3.1.0,>=3.0.5->en-ner-eco-md==1.0.0) (3.1)\n",
      "Requirement already satisfied: urllib3<1.27,>=1.21.1 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from requests<3.0.0,>=2.13.0->spacy<3.1.0,>=3.0.5->en-ner-eco-md==1.0.0) (1.26.6)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from requests<3.0.0,>=2.13.0->spacy<3.1.0,>=3.0.5->en-ner-eco-md==1.0.0) (2021.5.30)\n",
      "Requirement already satisfied: charset-normalizer~=2.0.0 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from requests<3.0.0,>=2.13.0->spacy<3.1.0,>=3.0.5->en-ner-eco-md==1.0.0) (2.0.0)\n",
      "Requirement already satisfied: click<7.2.0,>=7.1.1 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from typer<0.4.0,>=0.3.0->spacy<3.1.0,>=3.0.5->en-ner-eco-md==1.0.0) (7.1.2)\n",
      "Requirement already satisfied: MarkupSafe>=2.0 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from jinja2->spacy<3.1.0,>=3.0.5->en-ner-eco-md==1.0.0) (2.0.1)\n",
      "Collecting https://github.com/nleguillarme/taxonerd/releases/download/v1.1.1/en_ner_eco_biobert-1.0.0.tar.gz\n",
      "  Downloading https://github.com/nleguillarme/taxonerd/releases/download/v1.1.1/en_ner_eco_biobert-1.0.0.tar.gz (402.1 MB)\n",
      "\u001b[K     |████████████████████████████████| 402.1 MB 113 kB/s \n",
      "\u001b[?25hRequirement already satisfied: spacy<3.1.0,>=3.0.5 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from en-ner-eco-biobert==1.0.0) (3.0.7)\n",
      "Requirement already satisfied: jinja2 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from spacy<3.1.0,>=3.0.5->en-ner-eco-biobert==1.0.0) (3.0.1)\n",
      "Requirement already satisfied: setuptools in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from spacy<3.1.0,>=3.0.5->en-ner-eco-biobert==1.0.0) (58.0.4)\n",
      "Requirement already satisfied: spacy-legacy<3.1.0,>=3.0.5 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from spacy<3.1.0,>=3.0.5->en-ner-eco-biobert==1.0.0) (3.0.8)\n",
      "Requirement already satisfied: murmurhash<1.1.0,>=0.28.0 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from spacy<3.1.0,>=3.0.5->en-ner-eco-biobert==1.0.0) (1.0.5)\n",
      "Requirement already satisfied: srsly<3.0.0,>=2.4.1 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from spacy<3.1.0,>=3.0.5->en-ner-eco-biobert==1.0.0) (2.4.1)\n",
      "Requirement already satisfied: thinc<8.1.0,>=8.0.3 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from spacy<3.1.0,>=3.0.5->en-ner-eco-biobert==1.0.0) (8.0.10)\n",
      "Requirement already satisfied: blis<0.8.0,>=0.4.0 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from spacy<3.1.0,>=3.0.5->en-ner-eco-biobert==1.0.0) (0.7.4)\n",
      "Requirement already satisfied: wasabi<1.1.0,>=0.8.1 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from spacy<3.1.0,>=3.0.5->en-ner-eco-biobert==1.0.0) (0.8.2)\n",
      "Requirement already satisfied: catalogue<2.1.0,>=2.0.4 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from spacy<3.1.0,>=3.0.5->en-ner-eco-biobert==1.0.0) (2.0.5)\n",
      "Requirement already satisfied: tqdm<5.0.0,>=4.38.0 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from spacy<3.1.0,>=3.0.5->en-ner-eco-biobert==1.0.0) (4.62.2)\n",
      "Requirement already satisfied: packaging>=20.0 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from spacy<3.1.0,>=3.0.5->en-ner-eco-biobert==1.0.0) (21.0)\n",
      "Requirement already satisfied: pydantic!=1.8,!=1.8.1,<1.9.0,>=1.7.4 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from spacy<3.1.0,>=3.0.5->en-ner-eco-biobert==1.0.0) (1.8.2)\n",
      "Requirement already satisfied: preshed<3.1.0,>=3.0.2 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from spacy<3.1.0,>=3.0.5->en-ner-eco-biobert==1.0.0) (3.0.5)\n",
      "Requirement already satisfied: cymem<2.1.0,>=2.0.2 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from spacy<3.1.0,>=3.0.5->en-ner-eco-biobert==1.0.0) (2.0.5)\n",
      "Requirement already satisfied: typer<0.4.0,>=0.3.0 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from spacy<3.1.0,>=3.0.5->en-ner-eco-biobert==1.0.0) (0.3.2)\n",
      "Requirement already satisfied: requests<3.0.0,>=2.13.0 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from spacy<3.1.0,>=3.0.5->en-ner-eco-biobert==1.0.0) (2.26.0)\n",
      "Requirement already satisfied: pathy>=0.3.5 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from spacy<3.1.0,>=3.0.5->en-ner-eco-biobert==1.0.0) (0.6.0)\n",
      "Requirement already satisfied: numpy>=1.15.0 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from spacy<3.1.0,>=3.0.5->en-ner-eco-biobert==1.0.0) (1.21.2)\n",
      "Requirement already satisfied: pyparsing>=2.0.2 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from packaging>=20.0->spacy<3.1.0,>=3.0.5->en-ner-eco-biobert==1.0.0) (2.4.7)\n",
      "Requirement already satisfied: smart-open<6.0.0,>=5.0.0 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from pathy>=0.3.5->spacy<3.1.0,>=3.0.5->en-ner-eco-biobert==1.0.0) (5.2.1)\n",
      "Requirement already satisfied: typing-extensions>=3.7.4.3 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from pydantic!=1.8,!=1.8.1,<1.9.0,>=1.7.4->spacy<3.1.0,>=3.0.5->en-ner-eco-biobert==1.0.0) (3.10.0.0)\n",
      "Requirement already satisfied: urllib3<1.27,>=1.21.1 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from requests<3.0.0,>=2.13.0->spacy<3.1.0,>=3.0.5->en-ner-eco-biobert==1.0.0) (1.26.6)\n",
      "Requirement already satisfied: idna<4,>=2.5 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from requests<3.0.0,>=2.13.0->spacy<3.1.0,>=3.0.5->en-ner-eco-biobert==1.0.0) (3.1)\n",
      "Requirement already satisfied: charset-normalizer~=2.0.0 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from requests<3.0.0,>=2.13.0->spacy<3.1.0,>=3.0.5->en-ner-eco-biobert==1.0.0) (2.0.0)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from requests<3.0.0,>=2.13.0->spacy<3.1.0,>=3.0.5->en-ner-eco-biobert==1.0.0) (2021.5.30)\n",
      "Requirement already satisfied: click<7.2.0,>=7.1.1 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from typer<0.4.0,>=0.3.0->spacy<3.1.0,>=3.0.5->en-ner-eco-biobert==1.0.0) (7.1.2)\n",
      "Requirement already satisfied: MarkupSafe>=2.0 in /Users/julianflowers/Library/r-miniconda/envs/spacy_condaenv/lib/python3.9/site-packages (from jinja2->spacy<3.1.0,>=3.0.5->en-ner-eco-biobert==1.0.0) (2.0.1)\n"
     ]
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "source": [
    "import taxonerd\n",
    "\n",
    "from taxonerd import TaxoNERD"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "source": [
    "ner = TaxoNERD(model=\"en_ner_eco_biobert\", prefer_gpu=False, with_abbrev=True)"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "source": [
    "ner.find_entities(\"Brown bears (Ursus arctos), which are widely distributed throughout the northern hemisphere, are recognised as opportunistic omnivore\")\n"
   ],
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>offsets</th>\n",
       "      <th>text</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>T0</th>\n",
       "      <td>LIVB 0 11</td>\n",
       "      <td>Brown bears</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T1</th>\n",
       "      <td>LIVB 13 25</td>\n",
       "      <td>Ursus arctos</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       offsets          text\n",
       "T0   LIVB 0 11   Brown bears\n",
       "T1  LIVB 13 25  Ursus arctos"
      ]
     },
     "metadata": {},
     "execution_count": 5
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "source": [
    "text = '''Genetic diversity encompasses the components of\n",
    "the genetic coding that structures organisms (nucleotides,\n",
    "genes, chromosomes) and variation in\n",
    "the genetic make-up between individuals within a\n",
    "population and between populations. This is the\n",
    "raw material on which evolutionary processes act.\n",
    "Perhaps themost basicmeasure of genetic diversity\n",
    "is genome size—the amount of DNA (Deoxyribonucleic\n",
    "acid) in one copy of a species’ chromosomes\n",
    "(also called the C-value). This can vary enormously,\n",
    "with published eukaryote genome sizes ranging\n",
    "between 0.0023 pg (picograms) in the parasitic microsporidium\n",
    "Encephalitozoon intestinalis and 1400\n",
    "pg in the free-living amoeba Chaos chaos (Gregory\n",
    "2008). These translate into estimates of 2.2 million\n",
    "and 1369 billion base pairs (the nucleotides on opposing\n",
    "DNA strands), respectively. Thus, even at\n",
    "this level the scale of biodiversity is daunting. Cell\n",
    "size tends to increase with genome size. Humans\n",
    "have a genome size of 3.5 pg (3.4 billion base pairs).\n",
    "Much of genome size comprises non-coding\n",
    "DNA, and there is usually no correlation between\n",
    "genome size and the number of genes coded. The\n",
    "genomes of more than 180 species have been\n",
    "completely sequenced and it is estimated that, for\n",
    "example, there are around 1750 genes for the bacteria\n",
    "Haemophilus influenzae and 3200 for Escherichia\n",
    "coli, 6000 for theyeast Saccharomyces cerevisiae, 19 000\n",
    "for the nematode Caenorhabditis elegans, 13 500 for\n",
    "the fruit fly Drosophila melanogaster, and\u000125 000 for\n",
    "the plantArabidopsis thaliana, themouseMusmusculus,\n",
    "brown rat Rattus norvegicus and human Homo\n",
    "sapiens. There is strong conservatism of some genes\n",
    "acrossmuchof thediversityof life.Thedifferences in\n",
    "genetic composition of species giveus indications of\n",
    "their relatedness,andthus important informationas\n",
    "to how the history and variety of life developed.\n",
    "Genes are packaged into chromosomes. The\n",
    "number of chromosomes per somatic cell thus\n",
    "far observed varies between 2 for the jumper ant\n",
    "Myrmecia pilosula and 1260 for the adders-tongue\n",
    "fern Ophioglossum reticulatum. The ant species reproduces\n",
    "by haplodiploidy, in which fertilized\n",
    "'''"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "source": [
    "ner.find_entities(text)"
   ],
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>offsets</th>\n",
       "      <th>text</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>T0</th>\n",
       "      <td>LIVB 597 625</td>\n",
       "      <td>Encephalitozoon intestinalis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T1</th>\n",
       "      <td>LIVB 1280 1302</td>\n",
       "      <td>Haemophilus influenzae</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T2</th>\n",
       "      <td>LIVB 1343 1351</td>\n",
       "      <td>theyeast</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T3</th>\n",
       "      <td>LIVB 1352 1376</td>\n",
       "      <td>Saccharomyces cerevisiae</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T4</th>\n",
       "      <td>LIVB 1393 1401</td>\n",
       "      <td>nematode</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T5</th>\n",
       "      <td>LIVB 1402 1424</td>\n",
       "      <td>Caenorhabditis elegans</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T6</th>\n",
       "      <td>LIVB 1441 1450</td>\n",
       "      <td>fruit fly</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T7</th>\n",
       "      <td>LIVB 1451 1474</td>\n",
       "      <td>Drosophila melanogaster</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T8</th>\n",
       "      <td>LIVB 1495 1520</td>\n",
       "      <td>plantArabidopsis thaliana</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T9</th>\n",
       "      <td>LIVB 1522 1541</td>\n",
       "      <td>themouseMusmusculus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T10</th>\n",
       "      <td>LIVB 1543 1552</td>\n",
       "      <td>brown rat</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T11</th>\n",
       "      <td>LIVB 1553 1570</td>\n",
       "      <td>Rattus norvegicus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T12</th>\n",
       "      <td>LIVB 1965 1975</td>\n",
       "      <td>jumper ant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T13</th>\n",
       "      <td>LIVB 1976 1993</td>\n",
       "      <td>Myrmecia pilosula</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T14</th>\n",
       "      <td>LIVB 2030 2054</td>\n",
       "      <td>Ophioglossum reticulatum</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T15</th>\n",
       "      <td>LIVB 2060 2063</td>\n",
       "      <td>ant</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            offsets                          text\n",
       "T0     LIVB 597 625  Encephalitozoon intestinalis\n",
       "T1   LIVB 1280 1302        Haemophilus influenzae\n",
       "T2   LIVB 1343 1351                      theyeast\n",
       "T3   LIVB 1352 1376      Saccharomyces cerevisiae\n",
       "T4   LIVB 1393 1401                      nematode\n",
       "T5   LIVB 1402 1424        Caenorhabditis elegans\n",
       "T6   LIVB 1441 1450                     fruit fly\n",
       "T7   LIVB 1451 1474       Drosophila melanogaster\n",
       "T8   LIVB 1495 1520     plantArabidopsis thaliana\n",
       "T9   LIVB 1522 1541           themouseMusmusculus\n",
       "T10  LIVB 1543 1552                     brown rat\n",
       "T11  LIVB 1553 1570             Rattus norvegicus\n",
       "T12  LIVB 1965 1975                    jumper ant\n",
       "T13  LIVB 1976 1993             Myrmecia pilosula\n",
       "T14  LIVB 2030 2054      Ophioglossum reticulatum\n",
       "T15  LIVB 2060 2063                           ant"
      ]
     },
     "metadata": {},
     "execution_count": 7
    }
   ],
   "metadata": {}
  }
 ],
 "metadata": {
  "orig_nbformat": 4,
  "language_info": {
   "name": "python",
   "version": "3.9.7",
   "mimetype": "text/x-python",
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "pygments_lexer": "ipython3",
   "nbconvert_exporter": "python",
   "file_extension": ".py"
  },
  "kernelspec": {
   "name": "python3",
   "display_name": "Python 3.9.6 64-bit ('spacy_condaenv': conda)"
  },
  "interpreter": {
   "hash": "195fb075c4f307fef8a4ed408ff4c564f2e8e412610cc5e11d8f9440dc84f368"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}